File: backend/components/metiers/threed/security/validators/input/input_validator.js
// input_validator.js – Validation des entrées

function validateInput(data) {
  if (typeof data !== 'object' || Array.isArray(data)) throw new Error('Entrée invalide : doit être un objet');
  if (!('user' in data) || !('action' in data)) throw new Error('Entrée invalide : champs requis manquants');
  return true;
}

module.exports = { validateInput };

File: backend/components/metiers/threed/security/validators/input/input_validator.py
# input_validator.py – Validation des entrées


def validate_input(data):
    # Exemple de validation avancée
    if not isinstance(data, dict):
        raise ValueError("Entrée invalide : doit être un dictionnaire")
    if "user" not in data or "action" not in data:
        raise ValueError("Entrée invalide : champs requis manquants")
    return True
